### PR TITLE
perf: Improve performance of stress relief

### PR DESCRIPTION
**Pull request by kentquirk**
## Which problem is this PR solving?

In testing stress relief, I found that it could turn off too easily, and it might then cause refinery to crash. 

## Short description of the changes

- Don't let stress relief turn off until stress levels are below the deactivation level for 10 consecutive seconds
- Move the calculation of stress relief into the periodic stress check, rather than doing it on every span. Now each span check needs only a read lock.
- Tweak the math for sigmoid (it had a little slop that was bugging me).

### CODE REVIEW

File: collect/stressRelief.go [hunk 1]
```diff
@@ -226,7 +226,7 @@ func (s *StressRelief) sigmoid(num, denom string) float64 {
 	r := s.ratio(num, denom)
 	// this is an S curve from 0 to 1, centered around 0.5 -- determined
 	// by messing around with a graphing calculator
-	stress := .395*math.Atan(6*(r-0.5)) + 0.5
+	stress := 0.400305589*math.Atan(6*(r-0.5)) + 0.5
```
Comment (TylerHelmuth) [inline collect/stressRelief.go:229]: Any value in this scalar being defined in a constant somewhere?

Comment (kentquirk) [reply collect/stressRelief.go:229]: I didn't really think so -- it's just designed to be an S-shaped curve from 0 to 1, and symmetric around 0.5. There's no reason to ever change it. Does it need more comments?

File: collect/stressRelief.go [hunk 2]
```diff
@@ -261,22 +261,17 @@ func (s *StressRelief) Recalc() {
...
 	s.reason = reason
-	s.lock.Unlock()
-}
 
-func (s *StressRelief) StressLevel() uint {
-	s.lock.RLock()
-	defer s.lock.RUnlock()
-	return s.stressLevel
-}
+	if s.stressLevel >= s.deactivateLevel {
```
[the remark this answers is not in the document]
Comment (kentquirk) [reply collect/stressRelief.go:269]: I put it out there deliberately, but on second thought, I can move it inside without causing problems, so I will.

File: collect/stressRelief.go (hunk 1 above)
Comment (TylerHelmuth) [reply collect/stressRelief.go:229]: Nah I just wanted to make sure it wouldn't get used to scale something else similarly in the future.

File: collect/stressRelief.go [hunk 3]
```diff
@@ -266,22 +268,24 @@ func (s *StressRelief) Recalc() {
...
 	case Monitor:
+		// If it's off, should we activate it?
 		if !s.stressed && s.stressLevel >= s.activateLevel {
 			s.stressed = true
 			s.Logger.Info().WithField("stress_level", s.stressLevel).WithField("reason", s.reason).Logf("StressRelief has been activated")
 		}
+		// We want make sure that stress relief is below the deactivate level
+		// for a minimum time after the last time we said it should be, so
+		// whenever it's above that value we push the time out.
+		if s.stressLevel >= s.deactivateLevel {
```
Comment (TylerHelmuth) [inline collect/stressRelief.go:285]: Do ww need to push out the time whenever `level > deactivateLimit` or only when currently stressed and `level > deactivateLimit`?

Comment (kentquirk) [reply collect/stressRelief.go]: I added an extra test for isStressed.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**